pyasdf/block.py

```diff
@@ -521,6 +528,46 @@ def compression(self, compression):
     def is_compressed(self):
         return self._compression is not None
 
+    @property
+    def checksum(self):
+        return self._checksum
+
+    def _set_checksum(self, checksum):
+        if checksum == b'\0' * 16:
+            self._checksum = None
+        else:
+            self._checksum = checksum
+
+    def _calculate_checksum(self, data):
+        m = hashlib.new('md5')
+        m.update(self.data)
```

Comment:
Just a thought--might be annoying to do right now depending how far along it has to pass checksum-related handling.  Maybe in the future it will be cool to add a high-level chunked write routine that can apply any number of filters in the process like checksum computing and compression :D